client/state/notices/middleware.js

```diff
@@ -0,0 +1,62 @@
+/**
+ * External dependencies
+ */
+import { translate } from 'i18n-calypso';
+import { truncate } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import { successNotice, errorNotice } from 'state/notices/actions';
+import { getSitePost } from 'state/posts/selectors';
+import {
+	POST_DELETE_FAILURE,
+	POST_DELETE_SUCCESS,
+	POST_SAVE_SUCCESS
+} from 'state/action-types';
+
+export const OBSERVERS = {
+	[ POST_DELETE_FAILURE ]: ( action, dispatch, getState ) => {
+		const post = getSitePost( getState(), action.siteId, action.postId );
+
+		let message;
+		if ( post ) {
+			message = translate( 'An error occurred while deleting "%s"', {
+				args: [ truncate( post.title, { length: 24 } ) ]
+			} );
+		} else {
+			message = translate( 'An error occurred while deleting the post' );
+		}
+
+		dispatch( errorNotice( message ) );
+	},
+	[ POST_DELETE_SUCCESS ]: ( action, dispatch ) => {
+		dispatch( successNotice( translate( 'Post successfully deleted' ) ) );
+	},
+	[ POST_SAVE_SUCCESS ]: ( action, dispatch ) => {
+		let text;
+		switch ( action.post.status ) {
+			case 'trash':
+				text = translate( 'Post successfully moved to trash' );
+				break;
+
+			case 'publish':
+				text = translate( 'Post successfully published' );
+				break;
+		}
+
+		if ( text ) {
+			dispatch( successNotice( text ) );
+		}
+	}
```

Comment:
> trouble following logic

It feels like we are just wired differently here, which is fine

> bounce between the function declaration and usage

Not sure what you are implying here. Isn't this one of the very reasons we use functions instead of writing one big flat source file? Maybe it's more about the extent to which we do this. I was struggling in this very same spot on my PR: where do I define inline and where do I separate. I feel like separating will be the need as these things get bigger (_ala_ the comments about file size) and it will allow us to use some reusable handling code (such as `showMessage`). While I don't care too much about small trees like this, I wonder about setting the precedent for others trying to learn what to do.

> difference between...

In the one case, we are testing a specific function for its behavior and that's all we need to care about. With the object wrapping around it, we're testing both the functionality _and_ the calling convention. We have to know about that convention and we have to make sure that we keep our tests in sync with any changes that come from changing the wrapper.

You could argue that this isn't a big deal, and I would probably concede that point. Nonetheless, I still find it useful to work towards a testing suite that doesn't tangle itself like that.